src/Aspire.Hosting.Garnet/GarnetBuilderExtensions.cs

```diff
@@ -0,0 +1,108 @@
+/// </summary>
+public static class GarnetBuilderExtensions
+{
+    /// <summary>
+    /// Adds a Garnet container to the application model.
+    /// </summary>
+    /// <remarks>
+    /// The default image is "garnet" and the tag is "1.0.2".
+    /// </remarks>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port to bind the underlying container to.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<GarnetResource> AddGarnet(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    {
+        var garnet = new GarnetResource(name);
+        return builder.AddResource(garnet)
+                      .WithEndpoint(port: port, targetPort: 6379, name: GarnetResource.PrimaryEndpointName)
+                      .WithImage(GarnetContainerImageTags.Image, GarnetContainerImageTags.Tag)
+                      .WithImageRegistry(GarnetContainerImageTags.Registry);
+    }
+
+    /// <summary>
+    /// Adds a named volume for the data folder to a Garnet container resource and enables Garnet persistence.
+    /// </summary>
+    /// <remarks>
+    /// Use <see cref="WithPersistence(IResourceBuilder{GarnetResource}, TimeSpan?, long)"/> to adjust Garnet persistence configuration, e.g.:
+    /// <code>
+    /// var cache = builder.AddGarnet("cache")
+    ///                    .WithDataVolume()
+    ///                    .WithPersistence(TimeSpan.FromSeconds(10), 5);
+    /// </code>
+    /// </remarks>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the application and resource names.</param>
+    /// <param name="isReadOnly">
+    /// A flag that indicates if this is a read-only volume. Setting this to <c>true</c> will disable Garnet persistence.<br/>
+    /// Defaults to <c>false</c>.
+    /// </param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<GarnetResource> WithDataVolume(this IResourceBuilder<GarnetResource> builder, string? name = null, bool isReadOnly = false)
+    {
+        builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/data", isReadOnly);
+        if (!isReadOnly)
+        {
+            builder.WithPersistence();
+        }
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to a Garnet container resource and enables Garnet persistence.
+    /// </summary>
+    /// <remarks>
+    /// Use <see cref="WithPersistence(IResourceBuilder{GarnetResource}, TimeSpan?, long)"/> to adjust Garnet persistence configuration, e.g.:
+    /// <code>
```

Comment:
Thanks for adding these doc comments. Can you put this code in an `<example />` section?

Reply:
Please, yes of course. Check if this is what you meant?